### ISSUE TITLE
Object selection inside repeat loops

### ISSUE THREAD

**Issue by PascalLadalle** [4ian/GDevelop#1487]
## Describe the bug
When nesting object creation repeat loops, actions applying to the last object of a loop are also applied to its higher-level parent.

Longer version: I'm creating objects using nested repeat loops to make patterns. 
I need to apply instance variables to all objects to ID and position them.
I'm having problems when nesting these repeat loops because actions for one object are sometimes applied to parent objects too.
![image](https://user-images.githubusercontent.com/25978830/75636076-fc995100-5c1b-11ea-8238-3af8d5a1ebbe.png)
The middle column should read 1 2 3 4 5 6 7 8 9, but when a loop ends, the variable of the parent object is updated along with the variable of the last child.
Dummy project attached.

[DebugSpawnRankVar.zip](https://github.com/4ian/GDevelop/files/4272734/DebugSpawnRankVar.zip)

**Comment by 4ian** [comment]
This looks like it's working as designed. You're creating something in the actions, then you are using a sub event that is repeated... but it's still not a reason for GDevelop to forget the object you created. If you want GDevelop to forget about an object, you have to tell it about your object in *a separate event*. Otherwise, GDevelop remembers about the object you created (or objects that passed conditions) from all the parent events (i.e: the events that are *containing* your event).

In other words: put your "create object" actions (and other actions maybe) in a sub event of the repeat. The one in blue in my screenshot:

![image](https://user-images.githubusercontent.com/1280130/75636431-9b22b380-5c16-11ea-92f3-19b11a06380a.png)

**Comment by PascalLadalle** [comment]
so... no _direct_ parent-child relationship to avoid selection conflicts? 
The fact that it seemed to apply to the last repeat only made me think it could be a bug, but I was only seeing the result of the last repeat because it overwrote the previous ones... 🤦‍♂ 
Thanks!

**Comment by 4ian** [comment]
> no direct parent-child relationship to avoid selection conflicts?

Child events are inheriting the selection from their parent (which are themselves inheriting from their own parent), but the selection made by an event is reset once the event is finished. So a child is not inheriting the selection from "uncle/aunt" events (i.e: events that are next to the parent) ;)
